tapir/shifts/views/views.py

```diff
@@ -309,3 +314,210 @@ def get(self, request, *args, **kwargs):
         call_command("run_freeze_checks")
         messages.info(request, _("Frozen statuses updated."))
         return super().get(request, args, kwargs)
+
+
+class SolidarityShiftUsed(LoginRequiredMixin, RedirectView):
+    def post(self, request, *args, **kwargs):
+        date = datetime.now()
+        tapir_user = get_object_or_404(TapirUser, pk=kwargs["pk"])
+        solidarity_shift = SolidarityShift.objects.filter(is_used_up=False)[0]
+        solidarity_shift.is_used_up = True
+        solidarity_shift.date_used = date
+        solidarity_shift.save()
+
+        ShiftAccountEntry(
+            user=tapir_user,
+            value=1,
+            date=date,
+            description="Solidarity shift received",
+            is_from_welcome_session=False,
+        ).save()
+
+        messages.info(
+            request, _("Solidarity Shift received. Account Balance increased by 1.")
+        )
+
+        return redirect(tapir_user.get_absolute_url())
+
+
+class SolidarityShiftGiven(LoginRequiredMixin, RedirectView):
+    def post(self, request, *args, **kwargs):
+        tapir_user = get_object_or_404(TapirUser, pk=kwargs["pk"])
+        shift_attendance = tapir_user.shift_attendances.filter(
+            is_solidarity=False, state=2
+        )[0]
```

Comment:
Here I would check if the user has a high enough balance and return a 400 if not.

Reply:
Same as above. Shouldn't be possible to happen as far as I can see it, but I'll implement it anyways